packages/page-staking/src/Nomination/useBalance.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState, useEffect } from 'react';
+import {useApi, useCall} from '@polkadot/react-hooks';
+import { DeriveBalancesAll } from '@polkadot/api-derive/types';
+import { formatNumber } from '@polkadot/util';
+import { Balance } from '@polkadot/types/interfaces/runtime';
+import BN from 'bn.js';
+import { formatBalance } from '@polkadot/util';
+
+// Known account we want to use (available on dev chain, with funds)
+
+export function useBalance (address?: string | null): string | null  {
+  const api = useApi();
+  const balancesAll = useCall<DeriveBalancesAll>(api.api.derive.balances.all as any, [address]);
+  return balancesAll ? formatNumber(balancesAll.freeBalance) : null
+}
+
+export function useBalanceClear (address?: string | null): Balance | null  {
+  const api = useApi();
+  const balancesAll = useCall<DeriveBalancesAll>(api.api.derive.balances.all as any, [address]);
+  return balancesAll ? balancesAll.availableBalance : null
+}
+
+export type WholeFeesType = {
+  wholeFees: Balance;
+  feesLoading: boolean;
+}
+
+/**
+ * Get fees for controller account to allow start and stop nomination, existentialDeposit
+ * @param {string} bondedAddress
+ * @param {string} senderAddress
+ * @param {Array<string>} validators
+ * @return {WholeFeesType} { wholeFees, feesLoading }
+ */
+
+export function useFees (bondedAddress?: string | null, senderAddress?: string | null, validators?: string[]): WholeFeesType  {
+  const [paymentFees, setPaymentFees] = useState<Balance | null>(null);
+  const [bondFees, setBondFees] = useState<Balance | null>(null);
+  const [feesLoading, setFeesLoading] = useState<boolean>(false);
+  const [wholeFees, setWholeFees] = useState<any>(null);
+  const [startNominationFees, setStartNominationFees] = useState();
+  const [stopNominationFees, setStopNominationFees] = useState();
+  const api = useApi();
+  const existentialDeposit = api.api.consts.balances.existentialDeposit;
+  // any amount to get fees
+  const si = formatBalance.findSi('-');
+  const TEN = new BN(10);
+  const basePower = formatBalance.getDefaults().decimals;
+  const siPower = new BN(basePower + si.power);
+  const amount = new BN(1000).mul(TEN.pow(siPower));
+
+  async function getPaymentFees(addr1: string, addr2: string) {
```

Comment:
Couple of issue -

1. Each of these functions will be re-created each render
2. There is no allowance here fo when these queries fail (and they will, it is inconsistently implemented across chains)